lib/lambda/update/updatePackage.ts

```diff
@@ -0,0 +1,127 @@
+import { response } from "libs/handler-lib";
+import { APIGatewayEvent } from "aws-lambda";
+import { getPackage } from "libs/api/package";
+import { produceMessage } from "libs/api/kafka";
+
+type ActionType = "update-id" | "update-values" | "delete";
+
+export const handler = async (event: APIGatewayEvent) => {
+  const topicName = process.env.topicName as string;
+
+  if (!topicName) {
+    throw new Error("Topic name is not defined");
+  }
+
+  if (!event.body) {
+    return response({
+      statusCode: 400,
+      body: { message: "Event body required" },
+    });
+  }
+  try {
+    // TODO: allow user to input title of the accordion
+    const { packageId, action, updatedFields, changeReason } =
+      typeof event.body === "string"
+        ? JSON.parse(event.body)
+        : (event.body as {
+            packageId: string;
+            action: ActionType;
+            updatedFields: object;
+          });
+
+    if (!packageId || !action) {
+      return response({
+        statusCode: 400,
+        body: { message: "Package ID and action are required" },
+      });
+    }
+
+    const packageResult = await getPackage(packageId);
+
+    if (!packageResult) {
+      return response({
+        statusCode: 404,
+        body: { message: "No record found for the given id" },
+      });
+    }
+
+    if (action === "delete") {
+      await produceMessage(
+        topicName,
+        packageId,
+        JSON.stringify({
+          id: packageId,
+          deleted: true,
+          isAdminChange: true,
+          adminChangeType: "delete",
+        }),
+      );
+    }
+
+    if (action === "update-id") {
+      await produceMessage(
+        topicName,
+        packageId,
+        JSON.stringify({
+          id: packageId,
+          isAdminChange: true,
+          origin: "mako",
+        }),
+      );
+      // delete/hide old record and create new one with new id but same values
+    }
+
+    if (action === "update-values") {
+      const areValidFields = Object.keys(updatedFields).every((fieldKey) => {
+        return fieldKey in packageResult._source;
+      });
+
+      let changeMadeText: string;
+
+      if (!areValidFields) {
+        return response({
+          statusCode: 500,
```

Comment:
`400` is a forbidden request. `500` is a server error. Is `validFields` dependent on what comes from the client?